### PR TITLE
subscription only mode refactorings

### PR DESCRIPTION
**Pull request by R-iskey**

### CODE REVIEW

File: src/screens/subscribe/index.tsx [hunk 1]
```diff
@@ -143,14 +143,19 @@ export const Subscribe = () => {
...
+                  width={'100%'}
+                  onClick={handleSubscribe}
+                  disabled={channelLoading || isWrongNetwork || !channelAddress}
+                  size={'lg'}
+                >
+                  Subscribe
+                </Button>
+                <Text align={'center'}>
+                  You will need to connect your wallet, then sign a message to subscribe
+                </Text>
```
Comment (nivo33) [inline src/screens/subscribe/index.tsx:157]: Lets remove this @idogoldd

File: src/screens/subscribe/components/ConnectWalletButtons.tsx [hunk 2]
```diff
@@ -45,8 +45,7 @@ const ConnectWalletButtons = (props: PropsT) => {
   return (
     <Flex alignItems={'center'} direction={'column'} gap={1.5}>
       <Text mb={1} size={'sm'} align={'center'}>
-        <strong>Connect your wallet & subscribe</strong>
-        <br /> This won&apos;t trigger a transaction or cost any gas fees
+        <strong>Connect your wallet to allow</strong>
```
Comment (nivo33) [inline src/screens/subscribe/components/ConnectWalletButtons.tsx:48]: this doesnt make sense, lets revert the change
@idogoldd

File: src/screens/onboarding/setupApps/index.tsx [hunk 3]
```diff
@@ -69,7 +69,7 @@ export const SetupApps = (props: { appsToConnect: MessagingApp[] }) => {
       <Flex width={'100%'} justifyContent={'space-between'} gap={1} mb={2}>
         {props.appsToConnect && (
           <Button
-            onClick={() => setRoute(Routes.SelectApps)}
+            onClick={() => setRoute(Routes.SelectCategories)}
```
Comment (nivo33) [inline src/screens/onboarding/setupApps/index.tsx:72]: this should only happen if the mode is `isSubscribeMode`. You can get this from the `EnvrionmentContext`

File: src/screens/onboarding/selectCategories/index.tsx [hunk 4]
```diff
@@ -37,6 +61,9 @@ export const SelectCategories = () => {
           Next
         </Button>
       </Flex>
+      <Text color={'secondary'} mb={1}>
+        You can change your preferences at any time
+      </Text>
```
Comment (nivo33) [inline src/screens/onboarding/selectCategories/index.tsx:66]: only in subscribe mode

File: src/screens/onboarding/selectCategories/index.tsx [hunk 5]
```diff
@@ -1,27 +1,51 @@
...
+          (userPref) => userPref.commsChannelTagId === category.id
+        );
+
+        if (userPref?.enabled) {
+          updatePreference(category.id, web2App.toLowerCase() as Web2AppLower, userPref);
+        }
+      });
+    });
+
+    setRoute(Routes.SetupApps);
```
Comment (nivo33) [inline src/screens/onboarding/selectCategories/index.tsx:48]: if(!isSubscribeMode) // leave behavior as it was
else // do everything you're doing here

File: src/screens/onboarding/selectCategories/index.tsx [hunk 6]
```diff
@@ -1,27 +1,51 @@
...
+      app === MessagingApp.Discord ? discordGuildUrl && discordToken : true
+    );
+
+    availableWeb2Apps.forEach((web2App) => {
+      messageCategories.forEach((category) => {
+        const userPref = user?.preferences?.find(
+          (userPref) => userPref.commsChannelTagId === category.id
+        );
+
+        if (userPref?.enabled) {
```
Comment (nivo33) [inline src/screens/onboarding/selectCategories/index.tsx:42]: check that its not already toggled for the specific app to to avoid unnecessary API calls

Comment (R-iskey) [reply src/screens/onboarding/selectCategories/index.tsx:42]: Don't understand what's exactly should I check?

Comment (nivo33) [reply src/screens/onboarding/selectCategories/index.tsx:48]: if (userPref?.enbaled & !userPref[web2App.toLowerCase()])

Comment (R-iskey) [reply src/screens/onboarding/selectCategories/index.tsx:48]: but, currently user unable to change the specific app, so all values is by default is false

**Comment by nivo33** [comment]
Title needs to be changed, & missing "Previous" button, see Figma
Screen Shot 2023-02-23 at 13 16 13

**Comment by nivo33** [comment]
This page needs to be changed as well (see figma)
image
